app/Jobs/PollForMediaWikiJobsJob.php

```diff
@@ -6,37 +6,44 @@
 use Illuminate\Support\Facades\Http;
 use Illuminate\Support\Facades\Log;
 use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Http\Client\Pool;
 
 class PollForMediaWikiJobsJob extends Job implements ShouldBeUnique
 {
     public $timeout = 3600;
     public function handle (): void
     {
         $allWikiDomains = Wiki::all()->pluck('domain');
-        foreach ($allWikiDomains as $wikiDomain) {
-            if ($this->hasPendingJobs($wikiDomain)) {
+        $responses = Http::pool(function (Pool $pool) use ($allWikiDomains) {
```

Comment:
This is mostly following this tutorial https://medium.com/@antoine.lame/laravel-asynchronous-and-concurrent-http-requests-9c8e3e6d36bd